### ISSUE TITLE
Add group tags for matrix_cl functions and new matrix type

### ISSUE THREAD

**Issue by SteveBronder** [stan-dev/math#2412]
## Description

I was looking through the doc site and realized for a lot of the OpenCL ops we don't have them grouped under the opencl module.

https://mc-stan.org/math/d5/de5/group__opencl.html

It would be nice to do the same thing for the new matrix type as well so we just have a full list of supported functions generated

#### Current Version:
v4.0.1

**Comment by bbbales2** [comment]
Oh yeah this would be nice. We just add these ingroup tags: https://github.com/stan-dev/math/blob/develop/stan/math/opencl/prim/divide.hpp#L10 at the start of functions? What about the functions that handle both varmat and matvar?

**Comment by SteveBronder** [comment]
Yep! What we could do is have tags for `Matrix` and `MatrixVar`. Then we can have a group of just all the matrix functions Stan supports
